lib/rspec/core/configuration.rb

```diff
@@ -870,7 +874,8 @@ def full_description
       # Adds a formatter to the set RSpec will use for this run.
       #
       # @see RSpec::Core::Formatters::Protocol
-      def add_formatter(formatter, output=output_stream)
+      def add_formatter(formatter, output=output_wrapper)
+        output = OutputWrapper.new(output) unless output == output_wrapper
```

Comment:
Is there a reason why this can't just set the output on the existing wrapper, and pass that through?